### PR TITLE
add samples for SCC v2 mute config rules

### PR DESCRIPTION
**Pull request by thokalavinod**
https://cloud.google.com/security-command-center/docs/reference/rest/v2/organizations.locations.muteConfigs

Fixes #<ISSUE-NUMBER>

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist
- [ ] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md)
- [ ] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#readme-file)
- [ ] **Tests** pass:   `nox -s py-3.9` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] **Lint** pass:   `nox -s lint` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [ ] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: securitycenter/snippets_v2/snippets_mute_config_v2.py [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+    """
+     Creates a mute configuration under a given scope that will mute
+     all new findings that match a given filter.
+     Existing findings will NOT BE muted.
+     Args:
+         parent_path: use any one of the following options:
+                      - organizations/{organization_id}
+                      - folders/{folder_id}
+                      - projects/{project_id}
+         mute_config_id: Set a unique id; max of 63 chars.
```
Comment (rsamborski) [inline securitycenter/snippets_v2/snippets_mute_config_v2.py:31]: `location_id` is missing from docstring

File: securitycenter/snippets_v2/snippets_mute_config_v2.py [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+     """
```
Comment (rsamborski) [inline securitycenter/snippets_v2/snippets_mute_config_v2.py:32]: Please add the docstring for the returned value

File: securitycenter/snippets_v2/snippets_mute_config_v2.py [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+def delete_mute_rule(parent_path: str, location_id: str, mute_config_name: str) -> None:
+    """
+    Deletes a mute configuration given its resource name.
+    Note: Previously muted findings are not affected when a mute config is deleted.
+    Args:
+         parent_path: use any one of the following options:
+                     - organizations/{organization_id}
+                     - folders/{folder_id}
+                     - projects/{project_id}
+        mute_config_name: Set a unique id; max of 63 chars.
```
Comment (rsamborski) [inline securitycenter/snippets_v2/snippets_mute_config_v2.py:73]: ditto: `location_id` is missing from docstring

File: securitycenter/snippets_v2/snippets_mute_config_v2.py [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+
+    mute_config = client.create_mute_config(request=request)
+    print(f"Mute rule created successfully: {mute_config.name}")
+    return mute_config
+
+# [END securitycenter_create_mute_config_v2]
+
+
+# [START securitycenter_delete_mute_config_v2]
+def delete_mute_rule(parent_path: str, location_id: str, mute_config_name: str) -> None:
```
Comment (rsamborski) [inline securitycenter/snippets_v2/snippets_mute_config_v2.py:64]: nit: for uniformity please update `mute_config_name` to `mute_config_id`

File: securitycenter/snippets_v2/snippets_mute_config_v2.py [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+
+    client.delete_mute_config(request)
+    print(f"Mute rule deleted successfully: {mute_config_name}")
+
+
+# [END securitycenter_delete_mute_config_v2]
+
+
+# [START securitycenter_get_mute_config_v2]
+def get_mute_rule(parent_path: str, location_id: str, mute_config_name: str) -> None:
```
Comment (rsamborski) [inline securitycenter/snippets_v2/snippets_mute_config_v2.py:90]: nit: for uniformity please update `mute_config_name` to `mute_config_id`

File: securitycenter/snippets_v2/snippets_mute_config_v2.py [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+    """
+    Retrieves a mute configuration given its resource name.
+    Args:
+        parent_path: use any one of the following options:
+                     - organizations/{organization_id}
+                     - folders/{folder_id}
+                     - projects/{project_id}
```
Comment (rsamborski) [inline securitycenter/snippets_v2/snippets_mute_config_v2.py:97]: ditto: `location_id` is missing from docstring

File: securitycenter/snippets_v2/snippets_mute_config_v2.py [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+    """
+    Listing mute configs at organization level will return all the configs
+    at the org, folder and project levels.
+    Similarly, listing configs at folder level will list all the configs
+    at the folder and project levels.
+    Args:
+        parent: Use any one of the following resource paths to list mute configurations:
+                - organizations/{organization_id}
+                - folders/{folder_id}
+                - projects/{project_id}
```
Comment (rsamborski) [inline securitycenter/snippets_v2/snippets_mute_config_v2.py:125]: ditto: `location_id` is missing from docstring

File: securitycenter/snippets_v2/snippets_mute_config_v2.py [hunk 8]
```diff
@@ -0,0 +1,183 @@
...
+    # List all Mute Configs present in the resource.
+    for mute_config in response:
+        print(mute_config.name)
+    return response
+
+# [END securitycenter_list_mute_configs_v2]
+
+
+# [START securitycenter_update_mute_config_v2]
+def update_mute_rule(parent_path: str, location_id: str, mute_config_name: str) -> None:
```
Comment (rsamborski) [inline securitycenter/snippets_v2/snippets_mute_config_v2.py:143]: nit: for uniformity please update `mute_config_name` to `mute_config_id`

File: securitycenter/snippets_v2/snippets_mute_config_v2.py [hunk 9]
```diff
@@ -0,0 +1,183 @@
...
+    """
+    Updates an existing mute configuration.
+    The following can be updated in a mute config: description, and filter/ mute rule.
+    Args:
+        parent: Use any one of the following resource paths to list mute configurations:
+                - organizations/{organization_id}
+                - folders/{folder_id}
+                - projects/{project_id}
+        mute_config_name: Specify the name of the mute config to delete.
```
Comment (rsamborski) [inline securitycenter/snippets_v2/snippets_mute_config_v2.py:152]: fix: this description is incorrect, due to couple issues:
 - it's an update operation, not deletion
 - you ask for the full format, but this param should be just the id as the full path is constructed based on `parent_path`, `location_id` and `mute_config_id` (see line 164)

File: securitycenter/snippets_v2/snippets_mute_config_v2_test.py [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+from google.api_core.exceptions import InternalServerError, NotFound, ServiceUnavailable
+from google.cloud import securitycenter_v2
+from google.cloud.securitycenter_v2.services.security_center.pagers import (
+    ListFindingsPager,
+)
+import pytest
+
+import snippets_mute_config_v2
+
+# TODO(developer): Replace these variables before running the sample.
```
Comment (rsamborski) [inline securitycenter/snippets_v2/snippets_mute_config_v2_test.py:31]:
```suggestion
```

nit: This line doesn't make sense in the test.

File: securitycenter/snippets_v2/snippets_mute_config_v2_test.py [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+PROJECT_ID = os.environ["GOOGLE_CLOUD_PROJECT"]
+ORGANIZATION_ID = os.environ["GCLOUD_ORGANIZATION"]
+GOOGLE_APPLICATION_CREDENTIALS = os.environ["GOOGLE_APPLICATION_CREDENTIALS"]
+
+
+@pytest.fixture
+def mute_rule():
+    mute_rule_create = f"random-mute-create-{uuid.uuid4()}"
+    mute_rule_update = f"random-mute-update-{uuid.uuid4()}"
+    resp_create = snippets_mute_config_v2.create_mute_rule(f"projects/{PROJECT_ID}","global",mute_rule_create)
```
Comment (rsamborski) [inline securitycenter/snippets_v2/snippets_mute_config_v2_test.py:41]: suggestion: We should have an explicit test of create_mute_rule. Right now we only use it during fixture, which can make failing test investigation more complex.

File: securitycenter/snippets_v2/snippets_mute_config_v2_test.py [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+
+
+@backoff.on_exception(
+    backoff.expo, (InternalServerError, ServiceUnavailable, NotFound), max_tries=3
+)
+def test_update_mute_rule(mute_rule):
+    response = snippets_mute_config_v2.update_mute_rule(
+        f"projects/{PROJECT_ID}","global",mute_rule.get('update')
+    )
+    assert response.name == mute_rule.get('update_resp').name
```
Comment (rsamborski) [inline securitycenter/snippets_v2/snippets_mute_config_v2_test.py:81]: nit: we should check if the description was actually updated.

File: securitycenter/snippets_v2/snippets_mute_config_v2_test.py [hunk 13]
```diff
@@ -0,0 +1,82 @@
...
+    mute_rule_update = f"random-mute-update-{uuid.uuid4()}"
+    resp_create = snippets_mute_config_v2.create_mute_rule(f"projects/{PROJECT_ID}","global",mute_rule_create)
+    resp_update = snippets_mute_config_v2.create_mute_rule(f"projects/{PROJECT_ID}","global",mute_rule_update)
+
+    yield {"create": mute_rule_create, "create_resp": resp_create, "update": mute_rule_update, "update_resp": resp_update}
+
+    snippets_mute_config_v2.delete_mute_rule(
+        f"projects/{PROJECT_ID}","global",mute_rule_create
+    )
+    snippets_mute_config_v2.delete_mute_rule(
```
Comment (rsamborski) [inline securitycenter/snippets_v2/snippets_mute_config_v2_test.py:49]: suggestion: We should have an explicit test of delete_mute_rule. Right now we only use it during fixture, which can make failing test investigation more complex.

File: securitycenter/snippets_v2/snippets_mute_config_v2_test.py [hunk 14]
```diff
@@ -0,0 +1,82 @@
...
+    assert response.name == mute_rule.get('create_resp').name
+
+
+
+@backoff.on_exception(
+    backoff.expo, (InternalServerError, ServiceUnavailable, NotFound), max_tries=3
+)
+def test_list_mute_rules(mute_rule):
+    response = snippets_mute_config_v2.list_mute_rules(f"projects/{PROJECT_ID}","global")
+    assert len(list(response))>0
```
Comment (rsamborski) [inline securitycenter/snippets_v2/snippets_mute_config_v2_test.py:71]: nit: can we check if the expected rules are on the list?

**Review by rsamborski — CHANGES_REQUESTED** [review]

File: securitycenter/snippets_v2/snippets_mute_config_v2_test.py (hunk 14 above)
Comment (thokalavinod) [reply securitycenter/snippets_v2/snippets_mute_config_v2_test.py]: addressed

File: securitycenter/snippets_v2/snippets_mute_config_v2_test.py (hunk 13 above)
Comment (thokalavinod) [reply securitycenter/snippets_v2/snippets_mute_config_v2_test.py:48]: addressed

File: securitycenter/snippets_v2/snippets_mute_config_v2_test.py (hunk 12 above)
Comment (thokalavinod) [reply securitycenter/snippets_v2/snippets_mute_config_v2_test.py]: addressed

File: securitycenter/snippets_v2/snippets_mute_config_v2_test.py (hunk 11 above)
Comment (thokalavinod) [reply securitycenter/snippets_v2/snippets_mute_config_v2_test.py:40]: addressed

File: securitycenter/snippets_v2/snippets_mute_config_v2_test.py (hunk 10 above)
Comment (thokalavinod) [reply securitycenter/snippets_v2/snippets_mute_config_v2_test.py]: addressed

File: securitycenter/snippets_v2/snippets_mute_config_v2.py (hunk 9 above)
Comment (thokalavinod) [reply securitycenter/snippets_v2/snippets_mute_config_v2.py]: addressed

File: securitycenter/snippets_v2/snippets_mute_config_v2.py (hunk 8 above)
Comment (thokalavinod) [reply securitycenter/snippets_v2/snippets_mute_config_v2.py]: addressed

File: securitycenter/snippets_v2/snippets_mute_config_v2.py (hunk 7 above)
Comment (thokalavinod) [reply securitycenter/snippets_v2/snippets_mute_config_v2.py:133]: addressed

File: securitycenter/snippets_v2/snippets_mute_config_v2.py (hunk 6 above)
Comment (thokalavinod) [reply securitycenter/snippets_v2/snippets_mute_config_v2.py:102]: addressed

File: securitycenter/snippets_v2/snippets_mute_config_v2.py (hunk 5 above)
Comment (thokalavinod) [reply securitycenter/snippets_v2/snippets_mute_config_v2.py]: addressed

File: securitycenter/snippets_v2/snippets_mute_config_v2.py (hunk 4 above)
Comment (thokalavinod) [reply securitycenter/snippets_v2/snippets_mute_config_v2.py]: addressed

File: securitycenter/snippets_v2/snippets_mute_config_v2.py (hunk 3 above)
Comment (thokalavinod) [reply securitycenter/snippets_v2/snippets_mute_config_v2.py]: addressed

File: securitycenter/snippets_v2/snippets_mute_config_v2.py (hunk 2 above)
Comment (thokalavinod) [reply securitycenter/snippets_v2/snippets_mute_config_v2.py:34]: addressed

File: securitycenter/snippets_v2/snippets_mute_config_v2.py (hunk 1 above)
Comment (thokalavinod) [reply securitycenter/snippets_v2/snippets_mute_config_v2.py:31]: addressed

**Review by rsamborski — APPROVED** [review]
/gcbrun

**Pull request merged**